### ISSUE TITLE
can't "Manage Implements"

### ISSUE THREAD

**Issue by tatsumasagc** [mysurvive/pf2e-thaum-vuln#37]
FVTT Version 11.302
PF2e Version 5.0.2
Exploit Vulnerability Version 0.11.1
Socketlib Version 1.0.13

When I as the GM or my Player clicks on "Manage Implements" nothing happens on screen and the following error shows in the Debug Console.

![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/7320132/ad565e3e-e77b-429c-9940-e55eadb65123)

**Comment by arthurtrumpet** [comment]
Same problem but different error for me.
When Manage Implement window appears OK, can drag the items in but when clicking Confirm Changes get the following error
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/84977997/f065b94d-679d-4bc8-80a9-727849ca5538)

**Comment by mysurvive** [comment]
I will take a look at it sometime this weekend. Sorry about this, I've been pretty busy this week.

**Comment by arthurtrumpet** [comment]
No rush at all. Thank you for doing the module. Thaum is an absolute mess of a class so having some automation is great.
Am having quite a few issues so all this might be specific to my setup, so don't worry if it's not fixable.

**Comment by mysurvive** [comment]
@tatsumasagc can you verify that you have an implement chosen in all of your implement features? 
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/16602055/e8764625-587d-489c-9b04-3df8ad7c8a96)

This error looks like you don't have one selected. Whether you do or don't have one selected, please delete the feature and re-add it (you might need to down-level and re-level)

@arthurtrumpet try selecting the thaum token and in console run
_token.actor.unsetFlag("pf2e-thaum-vuln", "selectedImplements")

**Comment by arthurtrumpet** [comment]
That worked. It now takes and puts the icon next to the implements. Thank you
Main issue now is the weaknesses apply to everyone and not just the Thaumaturge. When applying mortal weakness you can see it update on the NPC token with each damage type rolled. But if another party member rolls an attack with the same type it also applies the weakness.
Noticed in your animated examples that you are using Target Damage mod. Have installed that and now everything works OK. Only applying weakness for the Thaum or Shared Weakness target.

**Comment by mysurvive** [comment]
Strange, my dev environment doesn't have target damage anymore (I switched worlds shortly after I made those gifs) and I don't have that issue. I will look into it when I am free to see what's going on... TD shouldn't be affecting anything EV does.

**Comment by mysurvive** [comment]
I found the issue... it was dumb.

**Comment by tatsumasagc** [comment]
was this patched?
I updated to 0.11.2 but i am still getting an error when i click "manage Implements" 
although the error has changed slightly
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/7320132/6859caa3-b86f-439b-a7a2-1a3f88afb87c)

**Comment by mysurvive** [comment]
try the command I posted above in console, the flag is probably borked.

**Comment by ColonialAF** [comment]
I tried the command and it did not work. Version 0.11.2, and verified there's an implement in feats.

**Comment by ColonialAF** [comment]
Nevermind! I restarted the server and we're fine.

**Comment by tatsumasagc** [comment]
first and second implemements have been choosen.
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/7320132/2e9f4cf3-7f6b-432e-9729-7576fd52459a)

clicking Manage Implements causies this error
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/7320132/1a625478-e1f9-4b54-9e4d-4aab37ca09a8)

inputting the following into the console 
`_token.actor.unsetFlag("pf2e-thaum-vuln", "selectedImplements")`
shows this in console but nothing pops up
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/7320132/8c0b5cf2-22ea-4b38-9486-93087e5592ad)

**Comment by tatsumasagc** [comment]
ok so i think i figured out the problem that character was imported from pathbuilder2e using pathmuncher.

when i recreated the character from scratch with the same feats etc it worked fine

**Comment by mysurvive** [comment]
I was just getting around to responding to this
- Nothing is supposed to pop up when running that command in the console
- Glad it's working out for you! Yeah there are issues with importers in general. Best to stay away from them as a rule of thumb.

**Comment by mysurvive** [comment]
Do you have any errors

**Comment by MarcyRangel** [comment]
No error appears. Just... nothing happens when I click manage implements. Because of that I cant use the Exploit Vulnerability macro, cause the macro asks me to manage my implements first haha

**Comment by mysurvive** [comment]
I meant in the console, there should be an error when you click the button if it doesn't open.

**Comment by mysurvive** [comment]
Hey, don't stress it. Press f12 with foundry open, it will open the console (like in chrome). Press the manage implements button and let me know what the red stuff that pops up says.

**Comment by MarcyRangel** [comment]
Here! 
![20230629_220214](https://github.com/mysurvive/pf2e-thaum-vuln/assets/138117296/f83c66f0-9901-4dc7-b8c3-024f434be834)

**Comment by mysurvive** [comment]
very good! In that little box in the bottom of the console, type 
_token.actor.unsetFlag("pf2e-thaum-vuln", "selectedImplements")
Then try to click the button again

**Comment by MarcyRangel** [comment]
It didnt work, this error appears: 
![image(1)](https://github.com/mysurvive/pf2e-thaum-vuln/assets/138117296/4cb69717-ff10-47a8-990f-6b24abcac941)

**Comment by mysurvive** [comment]
Looks like you don't have the token selected, you have to make sure to select the token and then run that.

**Comment by MarcyRangel** [comment]
The token is selected, same error. I think I will just give up, I dont want to keep bothering you with this 😭😭 thank you so much for trying to help!

**Comment by mysurvive** [comment]
I will be home in a couple of hours with no serious commitments. If you want, I truly don't mind hopping into your world and taking a look when you're free. Feel free to DM me on discord: mysurvive

**Comment by MarcyRangel** [comment]
Here, with the token selected I pasted that line in the console and then appeared that "Promise <pending>" . Then, I clicked manage implements, and then another error appeared 
![20230630_144604](https://github.com/mysurvive/pf2e-thaum-vuln/assets/138117296/70993d49-07d4-4b66-bbd8-41858a024b45)

**Comment by MarcyRangel** [comment]
If you want to enter our world that would be cool! My GM discord is "felipedelbue" (I am a player, I have been talking here for him cause he dont have a github account haha). We dont speak english fluently so he will not be able to talk to you through call, only text, if that's ok

**Comment by MarcyRangel** [comment]
My GM said you guys found the problem!! it seems the portuguese translation was the cause hahaha I just want to thank you so much for your help and patience!! Really, thank you!

**Comment by DoctorDankovsky** [comment]
I also encountered a problem, maybe it's the translation:
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/79034968/d2c18e7c-5dd6-4f96-a6c5-80a1c4b0dc0e)

**Comment by DiceKnights** [comment]
Hi, firstly thank you so much for your work on this, it is a godsend for one of my players running a thaumaturge.   I'm also having issues with the manage implements button doing nothing when I click on it (which means we cant use the other macros as keeps saying need an implement set up through manage implements). 
Can confirm have implement (Lantern) set up under first implement and esoterica. 
Running latest Foundry V11 and Pf2e and Explort Vuln module.  English language.
Get the following error.
![image](https://github.com/mysurvive/pf2e-thaum-vuln/assets/139088216/101f23c6-18d9-475b-af7f-9c96fe38a5e8)
reading through previous fixes I tried putting _token.actor.unsetFlag("pf2e-thaum-vuln", "selectedImplements" in console after clicking and getting error.. but I just get a promise {,pending>}  and the same error when clicking the button again.

**Comment by mysurvive** [comment]
Try restarting foundry and see if it works. Someone said something similar the other day and restarting their server fixed it for whatever reason after they ran the command. Unfortunately a storm knocked out my internet so there isn't much I can do to help right now :(

**Comment by DiceKnights** [comment]
Thanks for replying so quickly despite your storm situation. I tried closing and restarting foundry but no effect. Still getting the same error as previously.  I did discover that if I turn off the Enforce Held Implement it allows me to use the exploit vulnerability macro again which is totally enough for us to run smoothly for now :D  Excited to see where you take the module in future. Would be cool to see some of those implement effects (like the lantern granting bonus to perception checks etc in a radius). I tried to have a go at doing that myself based on other effects, but wow, its a different world you and I live in lol, so I appreciate what you do :)
